core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/provider_life_cycle.go

```diff
@@ -115,16 +110,28 @@ func (p *logEventProvider) register(ctx context.Context, lpFilter logpoller.Filt
 	if err != nil {
 		return fmt.Errorf("failed to get latest block while registering filter: %w", err)
 	}
+	isNewPollerFilter := p.poller.HasFilter(lpFilter.Name)
+	if !isNewPollerFilter {
```

Comment:
I think we'll also need to check the contents of the filter here to see if it has changed even if it's not a new filter?